lib/order.rb

```diff
@@ -7,12 +13,70 @@ def initialize(id, products)
       @products = products
     end
 
+    # gets the total plus tax of all of the products
     def total
-      # TODO: implement total
-    end
+      product_total = 0
+      subtotal = 0
+      @products.each_value do |prices|
+        subtotal += prices
+      end
+      product_total = (subtotal * 0.075).round(2)+ subtotal
+      return product_total
 
+    end
+    # Adds new product to @product array
     def add_product(product_name, product_price)
-      # TODO: implement add_product
+
+      if @products.include?(product_name)
+
+        return false
+      else
+        @products[product_name]= product_price
+        return true
+      end
+      return products
+
+    end
+
+    def self.all
+      all_orders = []
```

Comment:
This method is a little overly complicated.  

`CSV.open.each` will loop through each row, and below where you have `all_orders.each` will also loop through each row.  So you could merge the two, eliminating the CSV.open loop and replace the top line of the 2nd with:

`CSV.open(FILE_NAME, 'r').each do |order|`

However what' you've got works and that's good.